runners/direct-java/src/main/java/org/apache/beam/runners/direct/SideInputContainer.java

```diff
@@ -239,11 +257,21 @@ public boolean isReady(final PCollectionView<?> view, final BoundedWindow window
           "calling get() on PCollectionView %s that is not ready in window %s",
           view,
           window);
-      // Safe covariant cast
-      @SuppressWarnings("unchecked") Iterable<WindowedValue<?>> values =
-          (Iterable<WindowedValue<?>>) viewContents.getUnchecked(PCollectionViewWindow.of(view,
-              window)).get();
-      return view.getViewFn().apply(values);
+      // Safe covariant cast since we know that the view only contains KVs.
+      @SuppressWarnings("unchecked") Iterable<KV<?, ?>> elements = Iterables.transform(
+          (Iterable<WindowedValue<KV<?, ?>>>) viewContents.getUnchecked(
+              PCollectionViewWindow.of(view, window)).get(),
+          new Function<WindowedValue<KV<?, ?>>, KV<?, ?>>() {
```

Comment:
Instead of storing the input Multimap's entries as an iterable and converting back, can we just store the multimap directly? If that's too large for this change, file a followup jira.

Reply:
The plan was to eventually improve these representations (or have runner authors make these changes), see https://issues.apache.org/jira/browse/BEAM-3080 for the DirectRunner/ULR.

This was just to get the change in since all the runners here just do a trivial multimap -> iterable -> multimap for all the types anyways.